c10/util/C++17.h

```diff
@@ -8,17 +8,6 @@
 #include <type_traits>
 #include <utility>
 
-#if !defined(__clang__) && !defined(_MSC_VER) && defined(__GNUC__) && \
-    __GNUC__ < 5
-#error \
-    "You're trying to build PyTorch with a too old version of GCC. We need GCC 5 or later."
-#endif
-
-#if defined(__clang__) && __clang_major__ < 4
-#error \
-    "You're trying to build PyTorch with a too old version of Clang. We need Clang 4 or later."
-#endif
-
```

Comment:
WE better keep those checks in place, otherwise one might run into https://github.com/pytorch/pytorch/issues/120020